Automattic/jetpack
# ThreatsDataViews: History fields preset

## Description
Adds custom fields when viewing `Active threats` and `History` (only in table view).

## Proposed changes
- Display relevant fields dependant on `dataFields` and `statusFilters`
   - When `current` filter is included in `statusFilters` show `auto-fix`
   - When `fixed`, `ignored`, or `historic` (`fixed` and `ignored`) show `first-detected`, `fixed-on`, and hide `auto-fix`
   - Always show `Status` so you can distinguish between active/historic threats when viewing all only for the paid view
      - This could be further conditional to be hidden when only viewing `current` threats?
      - We may also want to include this field in list view for better clarity
- Show a more descriptive `Auto-fix` item in actions drop down only in list view, removing it from table view in favour of the `Auto-fix` field

**Potential issues:**
- If a user has no active threats and views all (either by removing all or adding all status filters), the `auto-fix` column will display even though they are technically viewing only historical threats.
- If a user is in a filter state where fields are not included (eg. no filters applied), they cannot apply such fields (eg. `first-detected`) manually from the settings.

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1592

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Review ThreatsDataViews storybooks
* Checkout this Protect branch on Jurassic Tube/Ninja
* Ensure that the toggle between `Active threats` and `History` applies the correct fields preset
* Ensure modifying filters also applies the correct field presets

## Screenshots

Active threats:
![Screen Shot 2024-12-17 at 12 52 37](https://github.com/user-attachments/assets/17b4d36f-fab2-4f8d-9152-629b5706a959)

History:
![Screen Shot on 2024-12-17 at 12-53-04](https://github.com/user-attachments/assets/0b2f6811-6ff7-4194-9b06-7a23d54b93db)

Action drop-down auto-fix in list view (now removed in table view):
![Screen Shot on 2024-12-17 at 12-54-01](https://github.com/user-attachments/assets/05939273-cb96-4a93-81e8-708c0e50fb82)
